Automattic/sensei
# Run tests when changes are introduced on the trunk

Fixes #6235 

### Changes proposed in this Pull Request

* Add a Trunk workflow to run tests when changes are added
* Add a badge on the Readme to improve the visibility of the pipeline state
* Move the Build Docs and Build to be executed after all tests are green. 

### Context
I am adding workflow only for the trunk branch because it enables us to use the GitHub Badges to see properly the workflow state, without it we need to add badges by workflow.

**Example of Badge:** 

[![Trunk Workflow](https://github.com/Automattic/sensei/actions/workflows/trunk.yml/badge.svg)](https://github.com/Automattic/sensei/actions/workflows/trunk.yml)

You can see the steps that are part of this workflow [here](https://github.com/Automattic/sensei/actions/runs/3651122864) 

## Testing instructions
- You can do an empty commit on this branch using  `git commit --allow-empty` and check if the pipeline is running.
You can check if it is running 

<img width="500" alt="Screenshot 2022-12-08 at 19 02 51" src="https://user-images.githubusercontent.com/38718/206576859-3b4152be-b79f-45df-9b21-c60259d4bfd7.png">


**Extra Notes:**
I set the trunk workflow to be executed on this branch only for demo purposes, before merge I will remove it.
